### DISCUSSION TITLE
fontawsome battery level indicator - can't make it to show text/icon on the bar

### DISCUSSION THREAD

**Discussion by adammendyk** [qtile/qtile#3245] [Q&A] [answered]
Hi all!
Can you help me by explaining why my widget is not showing desired icon, please?
Code is here -> https://github.com/adammendyk/awsomeFontBattery/tree/qtile_specs_v02 <- the latest approach :)
When you enable the object and calls at the bottom they work fine but qtile bar shows only N/A.
Can't get any further by myself, sorry. I don't know what I'm missing.
Best,
adamm

**Comment by elParaguayo** [comment, marked as answer]
A few issues jump out:
 - You're mixing base classes in your inheritance. Your definition inherits `base._TextBox`, your `__init__` references `base.InLoopPollText` and `draw` uses `base.ThreadPoolText`. I'd use `base.ThreadPoolText` for all of these.
 - You only query the battery status once, during `__init__`. This should be part of the `poll` method so it's updated each time.

There may be other issues but I can't test until later this evening.

**Reply by adammendyk** [reply]
Thank you so much! Alive and kicking!
Take care,
adamm
